models/discordactions.js

```diff
@@ -114,32 +116,56 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
-const getNumberOfMemberForGroups = async (groups = []) => {
+/**
+ * Enriches group data with membership information for a given Discord ID.
+ *
+ * @param {string} discordId - The Discord ID of the user.
+ * @param {Array<object>} groups - Array of group objects to process.
+ * @returns {Promise<Array<object>>} - An array of group objects with enriched information.
+ */
+const enrichGroupDataWithMembershipInfo = async (discordId, groups = []) => {
   try {
     if (!groups.length) {
       return [];
     }
-    const roleIds = groups.map((group) => group.roleid);
 
+    const groupCreatorIds = groups.reduce((ids, group) => {
+      if (group.createdBy) {
+        ids.add(group.createdBy);
+      }
+      return ids;
+    }, new Set());
+
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
+
+    const roleIds = groups.map((group) => group.roleid);
     const snapshots = await memberRoleModel.where("roleid", "in", roleIds).get();
-    const roleCount = {};
 
+    const groupsToMemberMaps = [];
+    const roleCount = {};
     snapshots.forEach((doc) => {
-      const roleToMemberMapping = doc.data();
-
-      if (roleCount[roleToMemberMapping.roleid]) {
-        roleCount[roleToMemberMapping.roleid] += 1;
-      } else {
-        roleCount[roleToMemberMapping.roleid] = 1;
+      const groupToMemberMapping = doc.data();
+      if (groupToMemberMapping) {
+        groupsToMemberMaps.push(groupToMemberMapping);
+        roleCount[groupToMemberMapping.roleid] = (roleCount[groupToMemberMapping.roleid] ?? 0) + 1;
       }
     });
 
-    return groups.map((group) => ({
-      ...group,
-      memberCount: roleCount[group.roleid] || 0,
-    }));
+    const userGroupMembershipsIds = findMemberGroupIds(discordId, groupsToMemberMaps);
+
+    return groups.map((group) => {
```

Comment:
also would suggest dont return a map directly from you `forEach` rather store it in a variable and then return it becomes easier to test when you decide to make it a small unit

Reply:
Can you tell me more about this? 
It seems okay to me to return the result from the map operation, as I've seen it done in many other code examples. I also like this approach. I'm curious to learn more about the possible issue here because it feels like something that can be tested. 
To me, using the `forEach` loop and a variable seems like a longer way to do the same thing and does not improve testability as we will be returning the same thing in both cases. And I can be wrong here as well and hence would like to hear your thoughts on it.

 Thank you for your review.

Reply:
Hey @vvaibhavdesai 
Please let me know your thoughts on this.